release-tools/metabuild.sh

```diff
@@ -30,18 +30,31 @@ add_makefile_prog() {
 }
 
 add_object_files() {
-	( ${MAKEBSD} -V SRCS |					\
+	# Set up useful variables
+	OBJ=$(${MAKEBSD} -V SRCS |				\
 	    sed -e 's| cpusupport-config.h||' |			\
 	    tr ' ' '\n' |					\
-	    sed -E 's/.c$/.o/' |				\
-	    while read F; do
+	    sed -E 's/.c$/.o/' )
+	CPP_ARGS_FIXED="-std=c99 -DCPUSUPPORT_CONFIG_FILE=\"cpusupport-config.h\" -I${SUBDIR_DEPTH} -MM"
+	OUT_CC_BEGIN="\${CC} \${CFLAGS_POSIX} ${CFLAGS_HARDCODED}"
+	OUT_CC_MID="-I${SUBDIR_DEPTH} \${IDIRS} \${CPPFLAGS} \${CFLAGS}"
+
+	# Generate build instructions for each object
+	for F in $OBJ; do
 		S=`${MAKEBSD} source-${F}`
 		CF=`${MAKEBSD} cflags-${F}`
 		IDIRS=`${MAKEBSD} -V IDIRS`
-		echo `${CPP} ${S} -std=c99 -DCPUSUPPORT_CONFIG_FILE=\"cpusupport-config.h\" -I${SUBDIR_DEPTH} ${IDIRS} -MM -MT ${F}` \
-			| sed -e 's| \\ | |g'
-		echo "	\${CC} \${CFLAGS_POSIX} ${CFLAGS_HARDCODED} ${CF} -I${SUBDIR_DEPTH} \${IDIRS} \${CPPFLAGS} \${CFLAGS} -c ${S} -o ${F}"
-	    done ) >> $OUT
+		# Get the build instructions, then remove newlines, condense
+		# multiple spaces, remove line continuations, and replace the
+		# final space with a newline.
+		${CPP} ${S} ${CPP_ARGS_FIXED} ${IDIRS} -MT ${F} |	\
+		    tr '\n' ' ' |					\
+		    tr -s ' '	|					\
+		    sed -e 's| \\ | |g' |				\
+		    sed -e 's| $||g'
+		printf "\n"
```

Comment:
Last-minute change here: on linux, `sed -e 's| $|\n|g'` does what you'd expect.  But FreeBSD is stricter about `sed` being a line editor:
```
$ man sed
...
Sed Regular Expressions
...
     2.   The escape sequence \n matches a newline character embedded in the
          pattern space.  You cannot, however, use a literal newline character
          in an address or in the substitute command.
```

Simple enough to add a `printf "\n"` instead.

Reply:
And yes, I double-checked the identical-ness of `Makefile`s after making this change.